packages/block-library/src/navigation/edit/use-create-navigation-menu.js

```diff
@@ -57,14 +58,47 @@ export default function useCreateNavigationMenu( clientId ) {
 					);
 				} );
 			}
+
 			const record = {
 				title,
+				slug,
 				content: serialize( blocks ),
 				status: 'publish',
 			};
 
+			// wp_navigation entities are keyed by slug in Core Data.
+			// By default saveEntityRecord will apply an optimisation and assume that if the
+			// recordKey is included in the record being saved then this should be a "update"
+			// rather than a "save" request. Therefore it will issue PUT and append the recordKey
+			// to the request path. However this will fail as no record yet exists.
+			// The record being created (see above) optionally includes a slug. This is in order that
+			// any template hierarchy within which the Nav block was position can be persisted
+			// in the slug and retrieved for later usage.
+			// Due to the inclusion of slug in the record being created is it necessary to bypass
+			// saveEntityRecord's inbuilt "optimisations" around PUT vs POST and always issue a POST request.
+			const forceCreateMethod = 'POST';
+
+			// By default saveEntityRecord will apply an optimisation and assume that if the
+			// recordKey is included in the record being saved then this should be a "update"
+			// rather than a "save" request. This function strips the recordKey that is appended to
+			// request path to ensure the request can be routed to the POST handler on the REST API
+			// endpoint. Failured to do this results in a 404.
+			const stripRecordKeyFromRequestPath = ( path, recordKey ) =>
+				path.replace( recordKey, '' );
+
 			// Return affords ability to await on this function directly
-			return saveEntityRecord( 'postType', 'wp_navigation', record )
+			return saveEntityRecord( 'postType', 'wp_navigation', record, {
+				__unstableFetch: ( params ) => {
+					return apiFetch( {
+						...params,
+						path: stripRecordKeyFromRequestPath(
```

Comment:
I don't understand if we use our own fetch why the need to strip the recordKey then?

Reply:
Because sending a request to POST /wp/navigation/{{some-slug}} would fail. Only PUT requests have an endpoint which matches that. POST requests should be made to `wp/navigation`.